tools/lambda-promtail/lambda-promtail/s3_test.go

```diff
@@ -0,0 +1,160 @@
+				"month":         "01",
+				"region":        "us-east-1",
+				"src":           "my-loadbalancer",
+				"type":          "elasticloadbalancing",
+				"year":          "2022",
+			},
+			wantErr: false,
+		},
+		{
+			name: "s3_flow_logs",
+			args: args{
+				record: events.S3EventRecord{
+					AWSRegion: "us-east-1",
+					S3: events.S3Entity{
+						Bucket: events.S3Bucket{
+							Name: "elb_logs_test",
+							OwnerIdentity: events.S3UserIdentity{
+								PrincipalID: "test",
+							},
+						},
+						Object: events.S3Object{
+							Key: "my-bucket/AWSLogs/123456789012/vpcflowlogs/us-east-1/2022/01/24/123456789012_vpcflowlogs_us-east-1_fl-1234abcd_20180620T1620Z_fe123456.log.gz",
+						},
+					},
+				},
+			},
+			want: map[string]string{
+				"account_id":    "123456789012",
+				"bucket":        "elb_logs_test",
+				"bucket_owner":  "test",
+				"bucket_region": "us-east-1",
+				"day":           "24",
+				"key":           "my-bucket/AWSLogs/123456789012/vpcflowlogs/us-east-1/2022/01/24/123456789012_vpcflowlogs_us-east-1_fl-1234abcd_20180620T1620Z_fe123456.log.gz",
+				"month":         "01",
+				"region":        "us-east-1",
+				"src":           "fl-1234abcd",
+				"type":          "vpcflowlogs",
+				"year":          "2022",
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := getLabels(tt.args.record)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("getLabels() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("getLabels() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_parseS3Log(t *testing.T) {
+	type args struct {
+		ctx       context.Context
+		b         *batch
+		labels    map[string]string
+		obj       io.ReadCloser
+		filename  string
+		batchSize int
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "vpcflowlogs",
+			args: args{
+				batchSize: 1024, // Set large enough we don't try and send to promtail
+				filename:  "../testdata/vpcflowlog.log.gz",
+				ctx:       context.TODO(),
+				b: &batch{
+					streams: map[string]*logproto.Stream{},
+				},
+				labels: map[string]string{
+					"type": FLOW_LOG_TYPE,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "albaccesslogs",
+			args: args{
+				batchSize: 1024, // Set large enough we don't try and send to promtail
+				filename:  "../testdata/albaccesslog.log.gz",
+				ctx:       context.TODO(),
+				b: &batch{
+					streams: map[string]*logproto.Stream{},
+				},
+				labels: map[string]string{
+					"type": LB_LOG_TYPE,
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			batchSize = tt.args.batchSize
+			tt.args.obj, _ = os.Open(tt.args.filename)
```

Comment:
Can we not miss the error check here please?. Can lead to weird panics and brittle test on the long run IMO.

Reply:
Added an Error check and message to point out the problem